src/engine.cc

```diff
@@ -306,6 +309,12 @@ void EngineController::SetPosition(const std::string& fen,
   search_.reset();
 }
 
+std::string EngineController::GetCurrentPositionFen() {
+  if (tree_ != nullptr) {
+    return tree_->HeadPosition().GetFen();
+  }
+  throw Exception("Need a 'go nodes..' before returning fen position.");
```

Comment:
Actually we need `position startpos|fen` I think, and not go nodes.
I think error message like "No game in progress", "Game not started" etc would be more appropriate.

Reply:
Hm, the way the fen is looked up, the position should already have been processed through the engine. For some reason this set of commands does not work:

> position fen 8/2p5/3p4/KP5r/1R3p1k/8/4P1P1/8 w - - 1 1
> get fen
> error Need a 'go nodes..' before returning fen position.

but this does:

 
> position fen 8/2p5/3p4/KP5r/1R3p1k/8/4P1P1/8 w - - 1 1
> go nodes 1
> ...
> get fen
> 8/2p5/3p4/KP5r/1R3p1k/8/4P1P1/8 w - - 1 0